From e925516c86f227b0de08b262a01a2077062032f6 Mon Sep 17 00:00:00 2001
From: Honggang Li <honli@redhat.com>
Date: Thu, 18 Aug 2016 03:07:52 -0400
Subject: [PATCH 1/4] Replace hardcode path "/opt/opafm" with "/usr/lib/opa-fm"

RHEL and Fedora packaging rules do not allow install files in
the "/opt" directory. So, take "/usr/lib/opa-fm" as private
directory for opa-fm package.

Signed-off-by: Honggang Li <honli@redhat.com>
---
 CommonInstall/comp_qlgc_fm.pl            |  4 ++--
 CommonInstall/test_fm_install.sh         | 20 ++++++++++----------
 HSM_Lib/ibaccess_lib/cs_info_file.c      |  2 +-
 IbaTools/FastFabric/opahostsmanalysis.sh |  2 +-
 IbaTools/man/opafmconfigdiff.manPage     |  6 +++---
 IbaTools/opacabletest/opacabletest.sh    |  2 +-
 LINUX/opacapture.sh                      |  4 ++--
 TestTools/defs.exp                       |  2 +-
 TestTools/run_hsm_loop_test_fast.sh      |  2 +-
 fm_config/config_diff.sh                 |  6 +++---
 10 files changed, 25 insertions(+), 25 deletions(-)

diff --git a/CommonInstall/comp_qlgc_fm.pl b/CommonInstall/comp_qlgc_fm.pl
index 0050a20..42fd691 100755
--- a/CommonInstall/comp_qlgc_fm.pl
+++ b/CommonInstall/comp_qlgc_fm.pl
@@ -61,7 +61,7 @@ sub disable_autostart2_opafm()
 
 sub start_opafm
 {
-	start_utility($ComponentInfo{'opafm'}{'Name'}, "/opt/opafm/runtime", "sm", "opafm");
+	start_utility($ComponentInfo{'opafm'}{'Name'}, "/usr/lib/opa-fm/runtime", "sm", "opafm");
 }
 
 sub stop_opafm
@@ -187,7 +187,7 @@ sub uninstall_opafm
 	rpm_uninstall_list("any", "verbose", ( "opa-fm", "opa-fm-debuginfo") );
 	system("rm -rf $ROOT/opt/opa/.comp_opafm.pl");
 	system("rmdir -p $ROOT/opt/iba/fm_tools 2>/dev/null");  # remove only if empty
-	system("rm -rf $ROOT/opt/opafm");
+	system("rm -rf $ROOT/usr/lib/opa-fm");
 	$ComponentWasInstalled{'opafm'}=0;
 }
 
diff --git a/CommonInstall/test_fm_install.sh b/CommonInstall/test_fm_install.sh
index b9d992a..33b8051 100644
--- a/CommonInstall/test_fm_install.sh
+++ b/CommonInstall/test_fm_install.sh
@@ -48,7 +48,7 @@ check_default()
 {
 	diff /etc/sysconfig/opafm.xml-sample /etc/sysconfig/opafm.xml
 	[ $? != 0 ] && echo "FAIL: installed does not match sample"
-	diff /etc/sysconfig/opafm.xml-sample /opt/opafm/etc/opafm.xml
+	diff /etc/sysconfig/opafm.xml-sample /usr/lib/opa-fm/etc/opafm.xml
 	[ $? != 0 ] && echo "FAIL: sample does not match default"
 }
 
@@ -56,7 +56,7 @@ check_nodefault()
 {
 	diff /etc/sysconfig/opafm.xml-sample /etc/sysconfig/opafm.xml
 	[ $? != 0 ] || echo "FAIL: installed should not match sample"
-	diff /etc/sysconfig/opafm.xml-sample /opt/opafm/etc/opafm.xml
+	diff /etc/sysconfig/opafm.xml-sample /usr/lib/opa-fm/etc/opafm.xml
 	[ $? != 0 ] && echo "FAIL: sample does not match default"
 }
 
@@ -89,7 +89,7 @@ check_norpmnew
 echo "========================================================================="
 echo "Install with iview_fm.config matching 4.4 version"
 rm_config
-cp /opt/opafm/etc/iview_fm.config.4x /etc/sysconfig/iview_fm.config
+cp /usr/lib/opa-fm/etc/iview_fm.config.4x /etc/sysconfig/iview_fm.config
 ./INSTALL -a
 check_default
 check_iview
@@ -98,7 +98,7 @@ check_norpmnew
 echo "========================================================================="
 echo "Install with iview_fm.config matching 4.3 version"
 rm_config
-cp /opt/opafm/etc/iview_fm.config.4.3 /etc/sysconfig/iview_fm.config
+cp /usr/lib/opa-fm/etc/iview_fm.config.4.3 /etc/sysconfig/iview_fm.config
 ./INSTALL -a
 check_default
 check_iview
@@ -107,7 +107,7 @@ check_norpmnew
 echo "========================================================================="
 echo "Install with iview_fm.config matching 4.3.1 version"
 rm_config
-cp /opt/opafm/etc/iview_fm.config.4.3.1 /etc/sysconfig/iview_fm.config
+cp /usr/lib/opa-fm/etc/iview_fm.config.4.3.1 /etc/sysconfig/iview_fm.config
 ./INSTALL -a
 check_default
 check_iview
@@ -116,7 +116,7 @@ check_norpmnew
 echo "========================================================================="
 echo "Install with iview_fm.config matching 4.2.1.2.1 version"
 rm_config
-cp /opt/opafm/etc/iview_fm.config.4.2.1.2.1 /etc/sysconfig/iview_fm.config
+cp /usr/lib/opa-fm/etc/iview_fm.config.4.2.1.2.1 /etc/sysconfig/iview_fm.config
 ./INSTALL -a
 check_default
 check_iview
@@ -125,7 +125,7 @@ check_norpmnew
 echo "========================================================================="
 echo "Install with opafm.xml matching 4.4 version"
 rm_config
-cp /opt/opafm/etc/opafm.xml /etc/sysconfig/opafm.xml
+cp /usr/lib/opa-fm/etc/opafm.xml /etc/sysconfig/opafm.xml
 ./INSTALL -a
 check_default
 check_noiview
@@ -134,7 +134,7 @@ check_norpmnew
 echo "========================================================================="
 echo "Install with opafm.xml matching sample but not 4.4 version (eg. upgrade)"
 rm_config
-cp /opt/opafm/etc/opafm.xml /etc/sysconfig/opafm.xml
+cp /usr/lib/opa-fm/etc/opafm.xml /etc/sysconfig/opafm.xml
 echo "<!-- extra line -->" >> /etc/sysconfig/opafm.xml
 cp /etc/sysconfig/opafm.xml /etc/sysconfig/opafm.xml-sample
 ./INSTALL -a
@@ -145,7 +145,7 @@ check_norpmnew
 echo "========================================================================="
 echo "Install with modified opafm.xml"
 rm_config
-cp /opt/opafm/etc/opafm.xml /etc/sysconfig/opafm.xml
+cp /usr/lib/opa-fm/etc/opafm.xml /etc/sysconfig/opafm.xml
 echo "<!-- extra line -->" >> /etc/sysconfig/opafm.xml
 ./INSTALL -a
 check_nodefault
@@ -155,7 +155,7 @@ check_norpmnew
 echo "========================================================================="
 echo "Install with modified iview_fm.config"
 rm_config
-cp /opt/opafm/etc/iview_fm.config.4x /etc/sysconfig/iview_fm.config
+cp /usr/lib/opa-fm/etc/iview_fm.config.4x /etc/sysconfig/iview_fm.config
 echo "#extra line" >> /etc/sysconfig/iview_fm.config
 ./INSTALL -a
 check_nodefault
diff --git a/HSM_Lib/ibaccess_lib/cs_info_file.c b/HSM_Lib/ibaccess_lib/cs_info_file.c
index d73967e..2d3f776 100755
--- a/HSM_Lib/ibaccess_lib/cs_info_file.c
+++ b/HSM_Lib/ibaccess_lib/cs_info_file.c
@@ -35,7 +35,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #define INFO_FILE "/etc/sysconfig/opa/opafm.info"
 static uint8_t		buffer [256];			// cleaned up read_buffer
-uint8_t		hfm_install_dir [256] = "/opt/opafm";	// base directory where everything is
+uint8_t		hfm_install_dir [256] = "/usr/lib/opa-fm";	// base directory where everything is
 
 static Status_t	read_line		(FILE *fd_config);
 
diff --git a/IbaTools/FastFabric/opahostsmanalysis.sh b/IbaTools/FastFabric/opahostsmanalysis.sh
index cea0d64..17ffd76 100755
--- a/IbaTools/FastFabric/opahostsmanalysis.sh
+++ b/IbaTools/FastFabric/opahostsmanalysis.sh
@@ -178,7 +178,7 @@ do
 		# check SM health/running
 		mkdir -p $latest_dir
 
-		/opt/opafm/bin/fm_cmd smShowCounters > $latest.smstatus 2>&1
+		/usr/lib/opa-fm/bin/fm_cmd smShowCounters > $latest.smstatus 2>&1
 		r=$?
 
 		if [ $r != 0 ]
diff --git a/IbaTools/man/opafmconfigdiff.manPage b/IbaTools/man/opafmconfigdiff.manPage
index ad4463e..e696ac5 100644
--- a/IbaTools/man/opafmconfigdiff.manPage
+++ b/IbaTools/man/opafmconfigdiff.manPage
@@ -47,8 +47,8 @@ Names of the configuration files to be compared.
 
 .SH Example 
 .NL
-opafmconfigdiff /etc/sysconfig/opafm.xml /opt/opafm/etc/opafm.xml
+opafmconfigdiff /etc/sysconfig/opafm.xml /usr/lib/opa-fm/etc/opafm.xml
 .br
-opafmconfigdiff -f /etc/sysconfig/opafm.xml /opt/opafm/etc/opafm.xml
+opafmconfigdiff -f /etc/sysconfig/opafm.xml /usr/lib/opa-fm/etc/opafm.xml
 .br
-opafmconfigdiff #8217#8211;d -uw /etc/sysconfig/opafm.xml /opt/opafm/etc/ opafm.xml
+opafmconfigdiff #8217#8211;d -uw /etc/sysconfig/opafm.xml /usr/lib/opa-fm/etc/ opafm.xml
diff --git a/IbaTools/opacabletest/opacabletest.sh b/IbaTools/opacabletest/opacabletest.sh
index dbcb870..02e638c 100755
--- a/IbaTools/opacabletest/opacabletest.sh
+++ b/IbaTools/opacabletest/opacabletest.sh
@@ -204,7 +204,7 @@ get_fmconfig()
 {
 	FM_CONFIG_DIR=/etc/sysconfig
 	FM_CONFIG_FILE=$CONFIG_DIR/opafm.xml
-	IFS_FM_BASE=/opt/opafm # default
+	IFS_FM_BASE=/usr/lib/opa-fm # default
 	if [ -s $FM_CONFIG_DIR/opa/opafm.info ]
 	then
 	    # get IFS_FM_BASE
diff --git a/LINUX/opacapture.sh b/LINUX/opacapture.sh
index 2e5a70c..7e23cc5 100755
--- a/LINUX/opacapture.sh
+++ b/LINUX/opacapture.sh
@@ -281,10 +281,10 @@ then
 	cp -r /sys/module /$dir/sys 2>/dev/null 
 fi
 
-if [ -f /opt/opafm/bin/fm_capture ]
+if [ -f /usr/lib/opa-fm/bin/fm_capture ]
 then
     echo "Gathering Host FM Information ..."
-    (cd /$dir; /opt/opafm/bin/fm_capture)
+    (cd /$dir; /usr/lib/opa-fm/bin/fm_capture)
 fi
 
 if [ -f /usr/bin/opa_osd_dump ]
diff --git a/TestTools/defs.exp b/TestTools/defs.exp
index 3d3fd07..c9e90e6 100644
--- a/TestTools/defs.exp
+++ b/TestTools/defs.exp
@@ -38,7 +38,7 @@ set g_xml_file_loc {/etc/sysconfig/opafm.xml}
 set g_default_xml_file test_run_default.xml
 set ibtools_dir {/opt/iba/ib_tools}
 
-set g_sample_xml_file_loc {/opt/opafm/etc/opafm.xml}
+set g_sample_xml_file_loc {/usr/lib/opa-fm/etc/opafm.xml}
 #set g_sample_xml_file_loc [ append g_sample_xml_file_loc $g_xml_file_loc -sample ]
 set g_saved_xml_file_loc {}
 set g_saved_xml_file_loc [ append g_saved_xml_file_loc $g_xml_file_loc .saved ]
diff --git a/TestTools/run_hsm_loop_test_fast.sh b/TestTools/run_hsm_loop_test_fast.sh
index 28fb2d6..5235ed4 100644
--- a/TestTools/run_hsm_loop_test_fast.sh
+++ b/TestTools/run_hsm_loop_test_fast.sh
@@ -36,7 +36,7 @@ end_test=0
 inject=0
 redundancy=0
 length=0
-SM_DIAG_PATH=/opt/opafm/bin/fm_cmd
+SM_DIAG_PATH=/usr/lib/opa-fm/bin/fm_cmd
 MIN_ISL_REDUNDANCY=4
 PATH_LENGTH=4
 
diff --git a/fm_config/config_diff.sh b/fm_config/config_diff.sh
index 2e97e80..271b6b9 100755
--- a/fm_config/config_diff.sh
+++ b/fm_config/config_diff.sh
@@ -69,9 +69,9 @@ Usage_full()
 	echo "to two FM instances described by file1 and file2" >&2
 	echo " " >&2
 	echo "Examples:" >&2
-	echo "  $(basename $PROG_NAME) /etc/sysconfig/opafm.xml /opt/opafm/etc/opafm.xml" >&2
-	echo "  $(basename $PROG_NAME) -f /etc/sysconfig/opafm.xml /opt/opafm/etc/opafm.xml" >&2
-	echo "  $(basename $PROG_NAME) -d -uw /etc/sysconfig/opafm.xml /opt/opafm/etc/opafm.xml" >&2
+	echo "  $(basename $PROG_NAME) /etc/sysconfig/opafm.xml /usr/lib/opa-fm/etc/opafm.xml" >&2
+	echo "  $(basename $PROG_NAME) -f /etc/sysconfig/opafm.xml /usr/lib/opa-fm/etc/opafm.xml" >&2
+	echo "  $(basename $PROG_NAME) -d -uw /etc/sysconfig/opafm.xml /usr/lib/opa-fm/etc/opafm.xml" >&2
 	exit 0
 
 }
-- 
2.7.4

